src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -61,6 +61,37 @@ public static bool IsZipDeployment(this IEnvironment environment)
                 !string.IsNullOrEmpty(environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteRunFromPackage));
         }
 
+        public static bool IsAppServiceWindowsEnvironment(this IEnvironment environment)
+        {
+            return environment.IsAppServiceEnvironment()
+                && !environment.IsLinuxAppServiceEnvironment()
+                && !environment.IsLinuxContainerEnvironment();
+        }
+
+        public static bool IsCoreToolsEnvironment(this IEnvironment environment)
+        {
+            return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(CoreToolsEnvironment));
+        }
+
+        public static bool IsPersistentFileSystemAvailable(this IEnvironment environment)
+        {
+            return environment.IsAppServiceWindowsEnvironment()
+                || environment.IsLinuxAppServiceEnvWithPersistentStorage()
+                || environment.IsCoreToolsEnvironment();
+        }
+
+        public static bool IsLinuxAppServiceEnvWithPersistentStorage(this IEnvironment environment)
+        {
+            if (environment.IsLinuxAppServiceEnvironment())
+            {
+                string storageConfig = environment.GetEnvironmentVariable(AzureAppServiceStorage);
+                bool isStorageSettingPresent = !string.IsNullOrEmpty(storageConfig);
```

Comment:
What is the expected behavior if setting AzureAppServiceStorage is not present? This method should return false?